app/models/vm_or_template/right_sizing.rb

```diff
@@ -168,16 +168,8 @@ def cpu_usagemhz_rate_average_max_over_time_period
   end
 
   def derived_memory_used_max_over_time_period
-    if has_attribute?("derived_memory_used_max_over_time_period")
-      self["derived_memory_used_max_over_time_period"]
-    else
-      opts = {
-        :end_date => Time.now.utc,
-        :days     => Metric::LongTermAverages::AVG_DAYS
-      }
-      VimPerformanceAnalysis.find_perf_for_time_period(self, "daily", opts)
-                            .maximum(:derived_memory_used)
-    end
+    perfs = VimPerformanceAnalysis.find_perf_for_time_period(self, "daily", :end_date => Time.now.utc, :days => Metric::LongTermAverages::AVG_DAYS)
+    perfs.collect(&:abs_max_derived_memory_used_value).compact.max
```

Comment:
What about the old code was causing it to not pick the maximum value?

Reply:
The old code was getting the max of daily averages, not the realtime peaks/troughs which were lost during rollup. (The burst-rollup part change is now determining and storing these during rollup time)